noir_stdlib/src/ec/tecurve.nr

```diff
@@ -227,34 +227,16 @@ mod curvegroup {
 
         // Check for equality
         fn eq(self, p: Point) -> bool {
-            if self.is_zero() == true {
-                p.is_zero()
-            } else if p.is_zero() == true {
-                false
-            } else {
-                let Self {x: x1, y: y1, t: _t1, z: z1} = self;
-                let Self {x: x2, y: y2, t: _t2, z:z2} = p;
-                
-                if x1*z2 == x2*z1 {
-                    y1*z2 == y2*z1
-                } else {
-                    false
-                }
-            }
+            let Self {x: x1, y: y1, t: _t1, z: z1} = self;
+            let Self {x: x2, y: y2, t: _t2, z:z2} = p;
+
+            (x1*z2 == x2*z1) & (y1*z2 == y2*z1)
         }
 
         // Check if zero
         fn is_zero(self) -> bool {
-            let Self {x, y, t, z} = self;
-            if y == z {
-                if x == t {
-                    x == 0
-                } else {
-                    false
-                }
-            } else {
-                false
-            }
+            let Self {x, y, t: _t, z} = self;
+            (x == 0) & (y == z)
```

Comment:
I would keep the `t==0` check here to avoid false positives outside of the curve.

Reply:
The reason I removed it was to make `self.is_zero()` and `self.into_affine().is_zero()` consistent. I'll add it back